core/Db/Schema/Mariadb.php

```diff
@@ -44,4 +51,45 @@ public function isOptimizeInnoDBSupported(): bool
         $semanticVersion = strstr($version, '-', $beforeNeedle = true);
         return version_compare($semanticVersion, '10.1.1', '>=');
     }
+
+    public function hasReachedEOL(): bool
+    {
+        $currentVersion = $this->getVersion();
+
+        // End of security update for certain MariaDb versions as of https://mariadb.org/about/#maintenance-policy
+
+        // Support for 10.6 LTS ends on 6th July 2026
+        if (
+            version_compare($currentVersion, '10.6', '>=') &&
+            version_compare($currentVersion, '10.7', '<') &&
+            Date::today()->isEarlier(Date::factory('2026-07-07'))
+        ) {
+            return false;
+        }
+
+        // Support for 10.11 LTS ends on 16th February 2028
+        if (
+            version_compare($currentVersion, '10.11', '>=') &&
+            version_compare($currentVersion, '10.12', '<') &&
+            Date::today()->isEarlier(Date::factory('2028-02-17'))
+        ) {
+            return false;
+        }
+
+        // Support for 11.4 LTS ends on 29th May 2029
+        if (
+            version_compare($currentVersion, '11.4', '>=') &&
+            version_compare($currentVersion, '11.5', '<') &&
+            Date::today()->isEarlier(Date::factory('2029-05-30'))
+        ) {
+            return false;
+        }
+
+        // Support for all versions prior 11.8 already ended
```

Comment:
Took me a while to realise this wasn't meant to be 10.6 because all other versions between 10.6 and 11.8 are covered by conditions above, so might be good to tweak the comment a bit
```suggestion
        // Support for all versions prior to 11.8 (not covered by conditions above) already ended
```